### PR TITLE
Create system testing framework #18

### PR DESCRIPTION
**Pull request by daniel-hunt-tessella**
Basic system testing framework ready for review

### CODE REVIEW

File: systemTests/test-scenarios/given/given_steps.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        self._ditto_api_server = DittoApiServer(context)
+
+    def _write_test_file(self, file_name, content):
+        with open(os.path.join(self._context.execution_folder_path, 'testing_area/src', file_name), 'w') as file:
+            file.write(content)
+
+    def simple_test_file_is_setup(self):
+        self._write_test_file('testA.txt', 'example test content A')
+
+    def an_ftp_client_is_connected(self):
```
Comment (robert-clegg-tessella) [inline systemTests/test-scenarios/given/given_steps.py]: Do we need this for DITTO?

File: ditto_web_api/requirements.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 flake8
 minio
 mock
+psutil
```
[the remark this answers is not in the document]
Comment (daniel-hunt-tessella) [reply ditto_web_api/requirements.txt]: Good spot - it was used in some gridftp stuff that is now commented out so I'll remove that

File: systemTests/test-scenarios/given/ditto_api_server.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+class DittoApiServer:
+  def __init__(self, context):
+    self._context=context
+
+  def is_started(self):
+    #self._context.gridftp_process = subprocess.Popen(['../run-gridftp-server.sh'],
+    #stdout=self._context.console_logger.stdout_log_writer,
+    #stderr=self._context.console_logger.stderr_log_writer,
+    #shell=True, cwd=r'/home/vagrant/src/build')
```
[the remark this answers is not in the document]
Comment (daniel-hunt-tessella) [reply systemTests/test-scenarios/given/ditto_api_server.py]: Yes it should! I was not sure what to replace it with so just left it

File: systemTests/test-scenarios/given/given_steps.py (hunk 1 above)
Comment (daniel-hunt-tessella) [reply systemTests/test-scenarios/given/given_steps.py]: Probably not - you know more about it than me!

Comment (robert-clegg-tessella) [reply systemTests/test-scenarios/given/given_steps.py]: Let's take it out

**Pull request merged**